packages/shared/src/types/PostCollectionConfiguration.ts

```diff
@@ -0,0 +1,98 @@
+import { useMemo } from 'react';
+
+import * as db from '../db';
+import * as ChannelAction from './ChannelActions';
+
+export interface PostCollectionConfiguration {
+  shouldMaintainVisibleContentPosition: boolean;
+
+  /**
+   * High-level type of collection - specificity is to indicate that e.g.
+   * `comy-list-top-to-bottom` is different from simply a inverted
+   * `compact-list-bottom-to-top`.
+   */
+  type: 'compact-list-bottom-to-top' | 'comfy-list-top-to-bottom' | 'grid';
```

Comment:
I went back and forth on this:
1. `columnCount` could be removed from `PostCollectionConfiguration`, replaced at call site with `columnCount = collectionConfig.type === 'grid' ? 2 : 1`. The same goes for pretty much any other item in `PostCollectionConfiguration`.
2. Another alternative: We could add dynamic dispatch to `type` (the type represented by `'compact-list-bottom-to-top' | 'comfy-list-top-to-bottom' | 'grid'` – let's call it `PostCollectionType`) so you could "ask" `columnCountFor(type: PostCollectionType): 1 | 2`/`useColumnCountFor(type: PostCollectionType): 1 | 2` – i.e. retaining a centralized set of features for `PostCollectionType`, but making the path there more flexible (you could use a hook to get the `columnCount`, you could pass other args, etc)